api/desecapi/tests/test_domains.py

```diff
@@ -303,6 +303,224 @@ def test_create_domains(self):
+$TTL 43200 ; 12 hours
+import-me.example IN SOA ns1.example.com. hostmaster.example.com. 2022021300 10800 3600 2419000 43200
+import-me.example NS ns1.example.com.
+www.import-me.example CNAME a.example.
+www.import-me.example A 127.0.0.1
+"""
+        name = 'import-me.example'
+        response = self.client.post(self.reverse('v1:domain-list'), {'name': name, 'zonefile': zonefile})
+        self.assertResponse(response, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(
+            response.json(),
+            {'zonefile': ['No other records with the same name are allowed alongside a CNAME record.']},
+        )
+
+    def test_create_domain_zonefile_import_name_non_apex_soa(self):
+        zonefile = """$ORIGIN .
+$TTL 43200 ; 12 hours
+asdf.import-me.example IN SOA ns1.example.com. hostmaster.example.com. 2022021300 10800 3600 2419000 43200
+import-me.example NS ns1.example.com.
+www.import-me.example CNAME a.example.
+www.import-me.example A 127.0.0.1
+"""
+        name = 'import-me.example'
+        response = self.client.post(self.reverse('v1:domain-list'), {'name': name, 'zonefile': zonefile})
+        self.assertResponse(response, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(
+            response.json(),
+            {'zonefile': [f'Zonefile includes an SOA record for a name different from {name}.']},
+        )
+
+    def test_create_domain_zonefile_import_syntax_error_line(self):
+        zonefile = """$ORIGIN .
+$TTL 43200 ; 12 hours
+import-me.example IN SOA ns1.example.com. hostmaster.example.com. 2022021300 10800 3600 2419000 43200
+import-me.example NS ns1.example.com.
+www.import-me.example CNAME a.example.
+www.import-me.example A asdf
+"""
+        name = 'import-me.example'
+        response = self.client.post(self.reverse('v1:domain-list'), {'name': name, 'zonefile': zonefile})
+        self.assertResponse(response, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(
+            response.json(),
+            {'zonefile': [f'Zonefile contains syntax error in line 6.']},
+        )
+
+    def test_create_domain_zonefile_import_foreign_rrset(self):
+        zonefile = f"""$ORIGIN .
+$TTL 43200 ; 12 hours
+import-me.example IN SOA ns1.example.com. hostmaster.example.com. 2022021300 10800 3600 2419000 43200
+import-me.example NS ns1.example.com.
+import-me.example A 127.0.0.1
+inject.{self.other_domain.name}. CNAME a.example.
+"""
+        name = 'import-me.example'
+        with self.assertPdnsRequests(
+                self.requests_desec_domain_creation(name, axfr=False, keys=False) +
+                self.requests_desec_rr_sets_update(name) +
+                [self.request_pdns_zone_retrieve_crypto_keys(name)]
+        ):
+            response = self.client.post(self.reverse('v1:domain-list'), {'name': name, 'zonefile': zonefile})
+        self.assertResponse(response, status.HTTP_201_CREATED)
+        self.assertRRsetDB(self.other_domain, subname='inject', type_='CNAME', rr_contents=set())
```

Comment:
Such records are ignored by the parser, not by us; we don't really an easy way of knowing if things were skipped. I've added this behavior to the docs. If dnspython ever changes this behavior it won't be a security problem for us because the following code uses an API that does not allow addition of records outside the zone.